tests/epochs.test.mjs

```diff
@@ -8,149 +8,279 @@ import {
+                distributed: {
+                  [STUB_OPERATOR_ADDRESS]: 16875675000, // received the full operator reward, but docked 25% for not observing
+                },
+              },
+            },
+          },
+          epochDistributionData,
+        );
+
+        // it should be pruned from state after distribution notice sent
+        const prunedEpoch = await getEpoch({
+          memory: tickMemory,
+          timestamp:
+            epochSettings.epochZeroStartTimestamp +
+            epochSettings.durationMs +
+            1,
+          epochIndex: 0, // try and get the pruned epoch by epoch index even though it has been pruned
+        });
+        assert.deepStrictEqual(prunedEpoch, null);
+
+        // the new epoch should be created and prescribed
+        const secondEpoch = await getEpoch({
+          memory: tickMemory,
+          timestamp:
+            epochSettings.epochZeroStartTimestamp + epochSettings.durationMs,
+        });
+        assert.deepStrictEqual(secondEpoch, {
+          epochIndex: 1,
+          hashchain: 'hashchain-'.padEnd(43, 'g'),
+          startTimestamp:
+            epochSettings.epochZeroStartTimestamp + epochSettings.durationMs,
+          endTimestamp:
+            epochSettings.epochZeroStartTimestamp +
+            epochSettings.durationMs * 2,
+          startHeight: 2,
+          arnsStats: {
+            totalActiveNames: 1,
+            totalGracePeriodNames: 0,
+            totalReservedNames: 0,
+            totalReturnedNames: 0,
+          },
+          prescribedObservers: [],
+          prescribedNames: [],
+          observations: {
+            failureSummaries: [],
+            reports: [],
+          },
+          prescribedObservers: [
+            {
+              observerAddress: STUB_ADDRESS,
+              gatewayAddress: STUB_OPERATOR_ADDRESS,
+              stakeWeight: 2.6875675,
+              gatewayPerformanceRatio: 1,
+              observerPerformanceRatio: 0.5,
+              compositeWeight: 5.375135,
+              normalizedCompositeWeight: 1,
+              tenureWeight: 4,
+              stake: INITIAL_OPERATOR_STAKE + 16875675000, // includes the new reward from the previous epoch
+              startTimestamp: 0,
+            },
+          ],
+          prescribedNames: ['prescribed-name'],
+          distributions: {
+            totalEligibleGatewayReward: 22493305945,
+            totalEligibleGateways: 1,
+            totalEligibleObserverReward: 2499256216,
+            totalEligibleRewards: 24992562162,
+            rewards: {
+              eligible: {
+                [STUB_OPERATOR_ADDRESS]: {
+                  delegateRewards: [],
+                  operatorReward: 24992562161, // 0.001 of the protocol balance after the transfers and name purchase
+                },
+              },
+            },
+          },
+        });
+        sharedMemory = tickMemory;
       });
     });
+    // TODO: add tests to create N epochs
```

Comment:
For this PR?

Reply:
i'll add a test to "skip" epochs for if we ever fall behind